### ISSUE TITLE
Duplicates Mistreated in `ad_stat` and `cvm_stat` -- FIXED

### ISSUE THREAD

**Issue by cdowd** [cdowd/twosamples#15]
This is a retrospective issue. The problem has been dealt with in release v1.2.0. 

This whole issue is derived from comments by Christian Fiedler (University of Toronto Stats). I greatly appreciate that assistance. 

## Summary

Technical challenges caused a code change that created a mismatch between the sampling routine and the test-statistic such that two test routines were over-powered. This has been fixed in the current release. 

## Basic problem

The CVM test and AD test both describe themselves as $\sum_{x \in k} |F(x)-E(x)|^p w(x)$  where $w(x)$ is some weighting function and $k$ is the joint sample. 

However, in the prior release (v1.1.1), these two test stats were calculated as $\sum_{unique x \in k} |F(x)-E(x)|^p w(x)$. Adding the restriction to unique values of x was implicit, and quasi-deliberate (see appendix: "Why unique Vals"). 

## Why is this an issue?

In a real sense, this is just a slightly different test statistic, and should not have a big impact.  But, because the resampling routine creates many more duplicates (resampling with replacement), the bootstrapped null distribution was implicitly summing over fewer observations -- making for lower output values. This meant that when calculating a p-value by comparing the test statistic to the bootstrapped test statistics, we were comparing the test statistic to artificially deflated values. This in turn made the p-values more significant than they ought to have been. 

This was a more substantial issue for smaller samples and for distributions that are non-discrete. Distributions with many pre-existing duplicates already have a limit on the number of summands and were relatively unaffected. 

## Fix

One proposed fix is to drop the equality check in the code. This causes other problems (see appendix: "Why unique Vals").

Instead, the actual fix creates a new counter that starts at 1 and increments for each duplicated value. When a value is finally added to the output, that value is multiplied by this duplicates counter -- thus counting the gap in ECDFs once for each observation of x.

## Appendix: Why unique Vals?

Simply put, the code operates by running through sorted observations and adding to the output for each observation. Observations that are duplicates are tricky because when the code is looking at them, the actual gap $|F(x)-E(x)|$ is not yet known -- it will be found once all the duplicated values are tallied. If we add to the sum at this time then there are two problems: 1) the code still isn't doing the test statistic -- but a different near relative. 2) the order in which duplicates wind up being sorted can matter, which means that the test statistic `ad_test(vec1,vec2)` will not equal `ad_test(vec2,vec1)`. Thus the code deliberately dropped all duplicates but one from the sum. However, this caused the underweighting problem discussed above.
